From cd059ecb349a848e670418a347c5f7c51e041ba4 Mon Sep 17 00:00:00 2001
From: fbrosson <fbrosson@localhost>
Date: Wed, 6 Jun 2018 20:38:08 +0000
Subject: Fix script in the test suite to better filter the output of nm.

src/check-symbols.sh compares the list of symbols in libharfbuzz.so against
a known list, $IGNORED_SYMBOLS, but Haiku has 2 symbols which are missing
in this list: _gSharedObjectHaiku{ABI,Version}.

diff --git a/src/check-symbols.sh b/src/check-symbols.sh
index d4eca50..b181814 100755
--- a/src/check-symbols.sh
+++ b/src/check-symbols.sh
@@ -7,7 +7,7 @@ test -z "$srcdir" && srcdir=.
 test -z "$libs" && libs=.libs
 stat=0
 
-IGNORED_SYMBOLS='_fini\|_init\|_fdata\|_ftext\|_fbss\|__bss_start\|__bss_start__\|__bss_end__\|_edata\|_end\|_bss_end__\|__end__\|__gcov_flush\|llvm_.*'
+IGNORED_SYMBOLS='_fini\|_init\|_fdata\|_ftext\|_fbss\|__bss_start\|__bss_start__\|__bss_end__\|_edata\|_end\|_bss_end__\|__end__\|__gcov_flush\|_gSharedObjectHaikuABI\|_gSharedObjectHaikuVersion\|llvm_.*'
 
 if which nm 2>/dev/null >/dev/null; then
 	:
-- 
2.17.1


From dd78e507d3f667f7aa2027fe719bb11735882e4f Mon Sep 17 00:00:00 2001
From: fbrosson <fbrosson@localhost>
Date: Wed, 6 Jun 2018 20:39:08 +0000
Subject: Patch src/check-libstdc++.sh to use objdump instead of ldd.


diff --git a/src/check-libstdc++.sh b/src/check-libstdc++.sh
index ce0bdab..ccfb420 100755
--- a/src/check-libstdc++.sh
+++ b/src/check-libstdc++.sh
@@ -10,10 +10,15 @@ stat=0
 
 if which ldd 2>/dev/null >/dev/null; then
 	LDD=ldd
+	GREP_NEEDED=cat
 else
 	# macOS specific tool
 	if which otool 2>/dev/null >/dev/null; then
 		LDD="otool -L"
+		GREP_NEEDED=cat
+	elif which objdump 2>/dev/null >/dev/null; then
+		LDD="objdump -x"
+		GREP_NEEDED="grep NEEDED"
 	else
 		echo "check-libstdc++.sh: 'ldd' not found; skipping test"
 		exit 77
@@ -29,7 +34,7 @@ for soname in harfbuzz harfbuzz-gobject; do
 		if ! test -f "$so"; then continue; fi
 
 		echo "Checking that we are not linking to libstdc++ or libc++ in $so"
-		if $LDD $so | grep 'libstdc[+][+]\|libc[+][+]'; then
+		if $LDD $so | $GREP_NEEDED | grep 'libstdc[+][+]\|libc[+][+]'; then
 			echo "Ouch, linked to libstdc++ or libc++"
 			stat=1
 		fi
-- 
2.17.1

